### ISSUE TITLE
Machine-readable (xpointer) locators on demand

### ISSUE THREAD

**Issue by angerhang** [brucemiller/LaTeXML#736]
We are trying to retire the KWARC fork of LaTeXML and the first primary difference we need to synchronize is the locator and I will just copy the comment that Deyan had on this: 

> We need to add a capability to latexml for switching to the stex-style
> (xpointer-style really) locators. Ideally it should be a trigger in the
> latexml.sty.ltxml binding.
> 
> For example, you can already do:
> 
> --preload=[ids]latexml.sty
> 
> to request adding ids to all XML nodes created by latexml. So I would
> imagine an option that allows for the xpointer-style locators as
> 
> --preload=[ids,xpointer_locators]latexml.sty
> 
> We can implement that internally by flipping a switch, that switches
> between latexml's default locators and our minimally modified stex ones
> (they're in Mouth.pm).

**Comment by dginev** [comment]
The very simple summary of this issue is:

We need machine-readable locators, when we write them down in attributes.

This is an sTeX use case so far, but it goes beyond that. Any web editor powered by latexml would benefit from machine-readable locator attributes, as it could then have fine-grained automatic connection between the source and rendered content. See #101 for the full discussion in that direction.

This issue can be seen as a prerequisite for #101, as making the current locators machine-readable is a low difficulty task on the way of making them as precise as possible.

**Comment by brucemiller** [comment]
Hi all; with #1013, we have basic support for structured locators (as opposed to the former strings), which also cover ranges; a `toAttribute` method on these objects turns them into XPath.  I'll ping @tkw1536 and/or @dginev to offer a short example of how to use them before I close.  Note also there's another open issue #101 relating to the accuracy of the locators which may need further work.

**Comment by tkw1536** [comment]
The newly merged PR does not fix #101 -- the locators are about as accurate as they were before. However, they are now accessible in a machine-readable format. 

One useful example would be to have the following code in a binding somewhere:

```perl
Tag('ltx:quote', afterClose => sub {
    my ($document, $node, $box) = @_;
    return unless $box;
    $node->setAttribute('srcref', $box->getLocator->toAttribute); });
```

Then any tex code like

```latex
\begin{quote}
some text here
\end{quote}
```

would end up with a srcref attribute somewhat like this: 

```xml
    <quote srcref="example.tex#textrange(from=9;0,to=11;11)">
      <p>some text here</p>
    </quote>
```

See [this gist](https://gist.github.com/tkw1536/7ca89a875e854232ea82abafb6b5a3d7) for a complete example.

**Comment by brucemiller** [comment]
I think this is closeable now, as well (with #101). Thanks all;
